docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,479 @@
+======= Saving the model to a file =======
+```
+
+Congratulations! You've now successfully built a machine learning model for recommending movies. You can find the source code for this tutorial at the [dotnet/samples](https://github.com/dotnet/samples/tree/master/machine-learning/tutorials/MovieRecommendation) repository.
+
+## Improve your model
+
+There are several ways that you can improve the performance of your model so that you can get more accurate predictions.
+
+### Data 
+
+Adding more training data that has enough samples for each user and movie id can help improve the quality of the recommendation model.
+
+[Cross validation](https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/train-cross-validation-ml-net) is a technique for evaluating models that randomly splits up data into subsets (instead of extracting out test data from the dataset like you did in this tutorial) and takes some of the groups as train data and some of the groups as test data. This method outperforms making a train-test split in terms of model quality.
+
+### Features
+
+In this tutorial, you only use the three `Features` (`user id`, `movie id`, and `rating`) that are provided by the dataset. 
+
+While this is a good start, in reality you might want to add other attributes or `Features` (e.g. age, gender, geo-location, etc.) if they are included in the dataset. Adding more relevant `Features` can help improve the performance of your recommendation model. 
+
+If you are unsure about which `Features` might be the most relevant for your machine learning task, you can also make use of Feature Contribution Calculation (FCC) and [Feature Permutation Importance](../how-to-guides/determine-global-feature-importance-in-model.md), which ML.NET provides to discover the most influential `Features`.
+
+### Algorithm
+
+While ML.NET provides good default training algorithms, you can further fine-tune performance by changing parameters on these individual algorithms.
+
+For `Matrix Factorization`, you can experiment with Learning Rate, ApproximationRank, NumberOfIterations, and several other options to see if that gives you better results.
+
+Here is an example of how to fine-tune the `Matrix Factorization Trainer` options:
+
+```CSharp
+var options = new MatrixFactorizationTrainer.Options
+{
+  MatrixColumnIndexColumnName = "userIdEncoded",
+  MatrixRowIndexColumnName = "movieIdEncoded",
+  LabelColumnName = "Label",
+  NumberOfThreads = 1,
+  ApproximationRank = 32,
+  LearningRate = 0.3
+};
+```
+
+### New user scenario
+One common problem in collaborative filtering is the cold start problem, which is when you have a new user with no previous data to draw inferences from. This is often solved by asking new users to create a profile and, for instance, rate movies they have seen in the past. While this puts some burden on the user, it provides some starting data for new users with no rating history.
+
+## Other Recommendation Engines
```

Comment:
Rename this to "Recommended resources "

Reply:
I particularly dislike resources, this makes it sound like we are giving some vanilla resources for user to look at. The intent here is quite different, we could say other scenarios, approaches or styles of recommendations which I believe it do justice to the content that follows.